src/cli-ux/action/base.ts

```diff
@@ -92,7 +92,7 @@ export class ActionBase {
     task.status = status
   }
 
-  public async pauseAsync(fn: () => Promise<any>, icon?: string) {
+  public async pauseAsync<T extends any>(fn: () => Promise<T>, icon?: string): Promise<T> {
```

Comment:
This will help with backwards compatability, in the case that typescript cannot infer T.